### PR TITLE
Update pcs to shm

### PR DESCRIPTION
**Pull request by kamilsa**
### Description of the Change

Updates pcs to use shared model

### Benefits

New model will be used in pcs

### Possible Drawbacks 

none

### CODE REVIEW

File: irohad/model/commit.hpp [hunk 1]
```diff
@@ -19,10 +19,11 @@
 #define IROHA_COMMIT_HPP
 
 #include <rxcpp/rx-observable.hpp>
-#include "model/block.hpp"
+#include "interfaces/iroha_internal/block.hpp"
 
 namespace iroha {
-  using Commit = rxcpp::observable<model::Block>;
+  using Commit = rxcpp::observable<std::shared_ptr<shared_model::interface::Block>>;
+  using OldCommit = rxcpp::observable<model::Block>;
```
Comment (kamilsa) [inline irohad/model/commit.hpp]: Used for compatibility with other components using old commit type

File: irohad/network/impl/peer_communication_service_impl.cpp [hunk 2]
```diff
@@ -27,18 +28,30 @@ namespace iroha {
...
+      ordering_gate_->propagate_transaction(
+          std::shared_ptr<model::Transaction>(transaction->makeOldModel()));
     }
 
-    rxcpp::observable<model::Proposal>
+    rxcpp::observable<std::shared_ptr<shared_model::interface::Proposal>>
     PeerCommunicationServiceImpl::on_proposal() {
-      return ordering_gate_->on_proposal();
+      return ordering_gate_->on_proposal().map(
+          [](auto prop) -> std::shared_ptr<shared_model::interface::Proposal> {
```
Comment (kamilsa) [inline irohad/network/impl/peer_communication_service_impl.cpp]: explicit return type definition used to cast returned value with proto type to interface type

File: shared_model/interfaces/base/signable.hpp [hunk 3]
```diff
@@ -88,6 +87,29 @@ namespace shared_model {
             and this->createdTime() == rhs.createdTime();
       }
 
+/**
```
Comment (kamilsa) [inline shared_model/interfaces/base/signable.hpp]: Note: copied from https://github.com/hyperledger/iroha/pull/972/files#diff-f5da1abc9499e5ab936bf6172f85dbb6R95.

File: irohad/network/peer_communication_service.hpp [hunk 4]
```diff
@@ -18,15 +18,17 @@
 #ifndef IROHA_PEER_COMMUNICATION_SERVICE_HPP
 #define IROHA_PEER_COMMUNICATION_SERVICE_HPP
 
-#include "model/block.hpp"
-#include "model/proposal.hpp"
+#include "interfaces/iroha_internal/block.hpp"
+#include "interfaces/iroha_internal/proposal.hpp"
```
Comment (x3medima17) [inline irohad/network/peer_communication_service.hpp]: You can use forward declaration

File: shared_model/interfaces/base/signable.hpp (hunk 3 above)
Comment (x3medima17) [reply shared_model/interfaces/base/signable.hpp]: Maybe lets wait until that code that code is accepted, before merging this PR?

File: test/integration/pipeline/tx_pipeline_integration_test_fixture.hpp [hunk 5]
```diff
@@ -100,11 +107,12 @@ class TxPipelineIntegrationTestFixture
...
   std::mutex m;
 
-  std::vector<iroha::model::Proposal> proposals;
-  std::vector<iroha::model::Block> blocks;
+  std::vector<std::shared_ptr<ProposalType>> proposals;
+  std::vector<std::shared_ptr<BlockType>> blocks;
 
-  using Commit = rxcpp::observable<iroha::model::Block>;
-  std::unique_ptr<TestSubscriber<iroha::model::Proposal>> proposal_wrapper;
+  using Commit = iroha::Commit;
```
Comment (x3medima17) [inline test/integration/pipeline/tx_pipeline_integration_test_fixture.hpp]: Do you really need it?

File: test/module/irohad/torii/torii_service_test.cpp [hunk 6]
```diff
@@ -256,12 +268,15 @@ TEST_F(ToriiServiceTest, StatusWhenBlocking) {
   }
 
   // create block from the all transactions but the last one
-  iroha::model::Block block;
-  block.transactions.insert(
-      block.transactions.begin(), txs.begin(), txs.end() - 1);
+  iroha::model::Block old_block;
```
Comment (x3medima17) [inline test/module/irohad/torii/torii_service_test.cpp]: Please use builders

File: shared_model/interfaces/base/signable.hpp (hunk 3 above)
Comment (kamilsa) [reply shared_model/interfaces/base/signable.hpp]: @Alexey-N-Chernyshov  can resolve this conflict when he will be merging his PR if he does not mind. What do you think, @Alexey-N-Chernyshov

Comment (Alexey-N-Chernyshov) [reply shared_model/interfaces/base/signable.hpp]: ok, i don't mind

File: irohad/network/impl/peer_communication_service_impl.cpp [hunk 7]
```diff
@@ -15,6 +15,7 @@ limitations under the License.
 */
 
 #include "network/impl/peer_communication_service_impl.hpp"
+#include "backend/protobuf/from_old_model.hpp"
```
Comment (luckychess) [inline irohad/network/impl/peer_communication_service_impl.cpp]: Please use absolute path (missed shared_model)

File: irohad/network/impl/peer_communication_service_impl.cpp [hunk 8]
```diff
@@ -27,18 +28,30 @@ namespace iroha {
...
-      ordering_gate_->propagate_transaction(transaction);
+      ordering_gate_->propagate_transaction(
+          std::shared_ptr<model::Transaction>(transaction->makeOldModel()));
     }
 
-    rxcpp::observable<model::Proposal>
+    rxcpp::observable<std::shared_ptr<shared_model::interface::Proposal>>
     PeerCommunicationServiceImpl::on_proposal() {
-      return ordering_gate_->on_proposal();
+      return ordering_gate_->on_proposal().map(
```
Comment (luckychess) [inline irohad/network/impl/peer_communication_service_impl.cpp]: Does map() call necessary here?

File: irohad/network/impl/peer_communication_service_impl.cpp [hunk 9]
```diff
@@ -27,18 +28,30 @@ namespace iroha {
...
+          [](auto prop) -> std::shared_ptr<shared_model::interface::Proposal> {
+            return std::make_shared<shared_model::proto::Proposal>(
+                shared_model::proto::from_old(prop));
+          });
     }
 
     rxcpp::observable<Commit> PeerCommunicationServiceImpl::on_commit() {
-      return synchronizer_->on_commit_chain();
+      return synchronizer_->on_commit_chain().map([](auto commit) -> Commit {
+        return commit.map(
```
Comment (luckychess) [inline irohad/network/impl/peer_communication_service_impl.cpp]: Same question as above.

File: irohad/network/impl/peer_communication_service_impl.cpp (hunk 7 above)
Comment (x3medima17) [reply irohad/network/impl/peer_communication_service_impl.cpp]: We can't :(

File: irohad/network/impl/peer_communication_service_impl.cpp (hunk 8 above)
Comment (kamilsa) [reply irohad/network/impl/peer_communication_service_impl.cpp]: Yes, `ordering_gate_->on_proposal()` returns observable on old model while this function should return observable on new model

File: irohad/model/commit.hpp [hunk 10]
```diff
@@ -19,10 +19,10 @@
 #define IROHA_COMMIT_HPP
 
 #include <rxcpp/rx-observable.hpp>
-#include "model/block.hpp"
+#include "interfaces/iroha_internal/block.hpp"
```
Comment (Alexey-N-Chernyshov) [inline irohad/model/commit.hpp]: #include old model please

**Pull request merged**